ArchiSteamFarm/MobileAuthenticator.cs

```diff
@@ -155,18 +155,18 @@ internal sealed class MobileAuthenticator {
 
 			await LimitConfirmationsRequestsAsync().ConfigureAwait(false);
 
-			HtmlDocument htmlDocument = await Bot.ArchiWebHandler.GetConfirmations(DeviceID, confirmationHash, time).ConfigureAwait(false);
+			IDocument htmlDocument = await Bot.ArchiWebHandler.GetConfirmations(DeviceID, confirmationHash, time).ConfigureAwait(false);
 
-			HtmlNodeCollection confirmationNodes = htmlDocument?.DocumentNode.SelectNodes("//div[@class='mobileconf_list_entry']");
+			List<IElement> confirmationNodes = htmlDocument?.SelectNodes("//div[@class='mobileconf_list_entry']");
 
-			if (confirmationNodes == null) {
+			if ((confirmationNodes == null) || (confirmationNodes.Count == 0)) {
```

Comment:
This entire segment was wrongly coded to begin with (my wrongdoing, before your changes). We should return `null` only when `htmlDocument` is null, and return empty collection when there are no confirmations. If you could improve this part, it'd be great, even if it isn't strict part of this PR (but you're already modifying those lines).

Basically move below `result = new HashSet<Confirmation>()` above (after `null` check against `htmlDocument`) and just `return result` if `confirmationNodes` is empty.